server/src/diagnostic-queue.ts

```diff
@@ -49,18 +49,23 @@ export class DiagnosticEventQueue {
     constructor(
         protected readonly publishDiagnostics: (params: lsp.PublishDiagnosticsParams) => void,
         protected readonly documents: LspDocuments,
-        protected readonly logger: Logger
+        protected readonly logger: Logger,
+        protected readonly ignoredDiagnosticCodes: number[],
     ) { }
 
     updateDiagnostics(kind: EventTypes, event: tsp.DiagnosticEvent): void {
         if (!event.body) {
             this.logger.error(`Received empty ${event.event} diagnostics.`)
             return;
         }
-        const { file } = event.body;
+        const { file, diagnostics: diagnosticsFromServer } = event.body;
+
+        // `d.code` can be undefined
+        const isDiagnosticAllowed = (d: tsp.Diagnostic) => !d.code || this.ignoredDiagnosticCodes.indexOf(d.code) === -1;
```

Comment:
`[].includes()` instead of `indexOf` maybe?

Reply:
Array.includes is supported since node 6 though so it won't be a problem.

Reply:
This project [currently targets](https://github.com/theia-ide/typescript-language-server/blob/master/tsconfig.settings.json#L7) es6 (or 2015), and it [looks like](https://github.com/microsoft/TypeScript/blob/master/src/lib/es2016.array.include.d.ts) `Array.includes` is only available for es7 (2016) compilation targets.

Reply:
I see. Then never mind me although that target should probably be updated in a separate PR. That target is very old.

Reply:
I guess targeting an older standard has the added benefit of maximizing compatibility, if for some reason you're stuck with an older node version.

Reply:
But ES7 is supported from Node7 (or Node 8 if you prefer "stable" release. It would be reasonable to require at least Node 8 as that is already 3 or 4 years old.

Reply:
Agreed that targeting the current [LTS version](https://nodejs.org/en/about/releases/) would be better.

Reply:
I'd lean toward getting the target bumped prior to this PR if either of you are interested. Based on https://nodejs.org/en/about/releases/ we should probably support Node v10 and up

Reply:
I agree bumping deps and updating the target JS version is high priority, but it's unrelated to the changes in this PR and shouldn't be a blocker for merging this imo.